src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -29,6 +29,13 @@ namespace ArborX
 namespace Details
 {
 
+enum SharedRadiiPolicy
+{
+  NONE,      // shared radius not updated
+  ATOMIC,    // shared radius updated atomically
```

Comment:
Considering that the "atomic" code path is not ever executed, I am wondering if that is worth the complexity added.  If we don't have it we don't need the enum and the `shared_radii` variable I am picking on below becomes a boolean whether to use shared radii or not.

Reply:
I think it is worth it. I think this needs to be revisited and analyzed some more, and the presence of this use case would clearly point us to the code without having to study it again. I did start with a boolean, but I decided to go to enum because of that. I also think in the longer term it may be worth introducing a command line switch for shared/non-shared that would require this option depending on the backend.

Reply:
OK. Backed off from implementing non-atomic path.